helix-core/src/graphemes.rs

```diff
@@ -5,7 +5,87 @@ use ropey::{iter::Chunks, str_utils::byte_to_char_idx, RopeSlice};
 use unicode_segmentation::{GraphemeCursor, GraphemeIncomplete};
 use unicode_width::UnicodeWidthStr;
 
-use std::fmt;
+use std::borrow::Cow;
+use std::fmt::{self, Debug, Display};
+use std::marker::PhantomData;
+use std::ops::Deref;
+use std::ptr::NonNull;
+use std::{slice, str};
+
+use crate::chars::{char_is_whitespace, char_is_word};
+use crate::LineEnding;
+
+pub fn tab_width_at(visual_x: usize, tab_width: u16) -> usize {
+    tab_width as usize - (visual_x % tab_width as usize)
+}
```

Comment:
Do we need to mark as `#[inline]` ?

Reply:
It's currently only used inside `graphemes.rs` so the `#[inline]` doesn't have any effect (as it only affects cross crate inlineing) but it's a good point regardless so I added the annotation in case it does get used in other crates in the future in [7f6882f](https://github.com/helix-editor/helix/pull/5420/commits/7f6882fe46afd89b8772fb4026d1500f98058636)